### ISSUE TITLE
BEES

### ISSUE THREAD

**Issue by gbromios** [team-soas/cornucopia#7]
# bees

apiculture needs to be broken into several abstract steps to be a fun part of minecraft (much like irl farming), and in short:

`hives` drop `bees`, put `bees` into an `apiary` and they'll make `honeycomb`
1. where do i get my bees
   - [x] `hive` blocks spawn randomly at worldgen somehow
   - [X] can be broken to drop some `honeycomb` and 1-2 `bee`, rarely a `queen`
   - [X] already have a nice looking model that should look good in the wild
2. where do i put my bees
   - [x] players can build an `apiary` block at a crafting table; the `apiary` provides a new "crafting interface"
     - slot for `bee` on the left, holds up to 64 worker bees
     - slot for the `queen` on the left
     - hexagonal slots on the right, where a `bee` might poop
   - [x] when the `apiary` updates, there's a chance based on A) the number of worker `bee`s and **B**) the number of flowers nearby; (also inversely proportional to the number of apiaries within 4 blocks!) that the apiary will try to fill one of its blocks with `wax`, `honeycomb`, and eventually `royal_jelly`, which can be used to make new `queens`.
   - [x] lastly, the `apiary` should look for nearby flowers and has a chance to clone one. this provides a farm-able source of flowers~✿ :D it also means you need at least one flower to start.
3. how do i make more of my bees
   ~~a lot of mods seem to dabble in "queen breeding" which is a little over the top for my taste. having drones just adds another useless inventory item to be collected and managed and its a mechanic that doesn't provide any return in and of itself.~~ i think i came up with a pretty good bee system here but i basically did the aforementioned with "royal jelly" instead of drones.
   - [x] if you have a queen in the apiary, you will get more bees. every sixteen bees requires a full honeycomb, otherwise the number of bees will decline to fit the honey level, down to a minimum of 8 with no honey :C. this means that you can only take 3 honey at a time if you want to sustain full bee numbers

( queen reproduction mechanics see #26 )
1. how do i access my honey, my bee syrup
   in short, put honeycomb into the presser and you'll get honey (and wax if we have wax in the mod). I think it can probably be eaten raw too but most recipes will need processed honey. production things will probably wait for another milestone
2. how should my bees look
   - [X] cute ~hopefully~

more specific areas for this milestone

this resource chain allows players to generate a small yet steady supply of honey without too much investment, but also allows them to gear it up by producing some queens, which can be farmed but not without some costs. #26 

also, i seem to remember the hives i saw on the last server looking very goofy as perfect 7x7 square blocks of flowers. our radii should be more organic and stop generating flowers above a certain density... see #27

lots of textures~
